src/github.com/getlantern/http-proxy/forward/forward.go

```diff
@@ -47,11 +47,7 @@ func New(opts *Options) filters.Filter {
 				return nil, err
 			}
 
-			idleConn := idletiming.Conn(conn, opts.IdleTimeout, func() {
-				if conn != nil {
-					conn.Close()
-				}
-			})
+			idleConn := idletiming.Conn(conn, opts.IdleTimeout, nil)
```

Comment:
The changes to http-proxy are being reviewed [here](https://github.com/getlantern/http-proxy/pull/84).